extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -151,16 +151,17 @@ private static Throwable logTenantConfigContextFailure(Throwable t, String tenan
         if (!oidcConfig.discoveryEnabled) {
             if (!isServiceApp(oidcConfig)) {
                 if (!oidcConfig.authorizationPath.isPresent() || !oidcConfig.tokenPath.isPresent()) {
-                    throw new OIDCException("'web-app' applications must have 'authorization-path' and 'token-path' properties "
-                            + "set when the discovery is disabled.");
+                    throw new ConfigurationException(
```

Comment:
If you pass in the 'configKeys' property to the constructor here you will get an editor on the error page in dev mode to allow you to fix it.

Reply:
@stuartwdouglas It is very cool, thanks, updated it. Note there are more `ConfigurationException`s in this code - but I was not sure yet about updating them, as some of them report an invalid combination, etc. I'll follow though a bit later and also add the OIDC config keys constants, I've added a few of them in #21693, [here](https://github.com/quarkusio/quarkus/pull/21693/files#diff-046bbd237257d7d7e787b03bffd235e4579119ca55f45953db69d1113c3bd8ccR16), so I'll move them may be to `OidcTenantConfig` and will reuse everywhere.